stdlib/public/Concurrency/AsyncStreamBuffer.swift

```diff
@@ -559,21 +568,25 @@ final class _AsyncStreamCriticalStorage<Contents>: @unchecked Sendable {
 
   var value: Contents {
     get {
-      lock()
-      let contents = _value
-      unlock()
-      return contents
+      self.withLock { $0 }
     }
 
     set {
-      lock()
-      withExtendedLifetime(_value) {
-        _value = newValue
-        unlock()
+      let oldValue = self.withLock {
+        let old = $0
+        $0 = newValue
+        return old
       }
+      extendLifetime(oldValue)
     }
   }
```

Comment:
This looks problematic. If we are ever doing an operation on `value` that is doing a `get` and then `set` after each other then we will acquire the lock twice and other code can interleave. I know you only added to this code but I think we need to change this altogether.

Reply:
you mean remove this 'convenience' pair of accessors entirely i presume?